argo.tf

```diff
@@ -1,4 +1,36 @@
 locals {
+  crds_argo_application_metadata = {
+    "labels" : try(var.crds_argo_metadata.labels, {}),
+    "annotations" : try(var.crds_argo_metadata.annotations, {}),
+    "finalizers" : try(var.crds_argo_metadata.finalizers, [])
+  }
+  crds_argo_application_values = {
+    "project" : var.argo_project
+    "source" : {
+      "repoURL" : var.helm_repo_oci ? local.helm_repo_url : "https://${local.helm_repo_url}"
+      "chart" : var.crds_helm_chart_name
+      "targetRevision" : var.helm_chart_version
+      "helm" : {
+        "releaseName" : "${var.helm_release_name}-crds"
+        "parameters" : [for k, v in var.crds_settings : tomap({ "forceString" : true, "name" : k, "value" : v })]
```

Comment:
Please fix parameter issues `"parameters" : length(var.settings) == 0 ? null : [for k, v in var.settings : tomap({ "forceString" : true, "name" : k, "value" : v })]`